### ISSUE TITLE
Unable to convert wallet to new schema

### ISSUE THREAD

**Issue by tejeskifly** [ciphrex/mSIGNA#35]
Hi,

I'm use msigna for long time ago, now i download the latest version and try to open my wallet, but i'm facing with a problme:

First msigna say:

"File was created using schema 17. Current schema version is 22.

Schema can be upgraded. Would you like to make a backup and migrate the file to the new schema now?"

So i choose OK for this.
Than I get this error:

"operation can only be performed in transaction"

Is there any way to convert my old wallet somehow? 

Thanks!

Regards
Daniel

**Comment by jnbordnrt** [comment]
I have the same issue. I upgraded and cannot access any accounts. 

Is there any user support for this product?

**Comment by MrFloip** [comment]
Same Issue here trying to convert from schema 16.
I cant seem to remember what Program I used to create the .vault file with.

**Comment by jnbordnrt** [comment]
Hi everyone,

I created a new vault file and then imported all my accounts. There are several issues with this:

1. It takes FORVER! It literally goes back to the day you created the vault and rebuilds it by importing all the blocks. It took me 10 days because the program keep quitting and I had to keep restarting it.
2. I managed to send a transaction during all of this with an older version but somehow the wallet sent the unspent output to a change address (this is normal) and that change address doesn't show up in my account address list. I actually did see it there in one of the versions but it was duplicated! 
3. When it sent the unspent output to this change address, the coins were immediately swept into another address which I believe to be a miner's address. I think I have lost those coins. It's a lot of money.

Suggestions:

1. When re-building the vault file, don't mess with anything
2. At points, stop the sync by disconnecting from the server and make a copy of the vault file. If anything goes wrong, you can restart it from that file and not have to start all over from scratch again

**Comment by jnbordnrt** [comment]
Does anyone here know how to obtain your private key from the BIP32 Extended Private Key? 

That will solve most of the issues because you can control your coins from another wallet or blockchain.info

**Comment by CodeShark** [comment]
There are some known issues with migrating vault files. DO NOT SEND transactions with v0.10.1 through 0.10.5 as these versions have a critical security issue.

Unfortunately, recreating the vault file from accounts is the only workaround.

**Comment by MSISGNAHELP** [comment]
Yes I have the same issue as above. I tried to import the old schema, and it says 

"error in vault file. Please import keychains into a new vault file and recreate account." But doing so doesn't give me access to my funds and I didn't export the account file. Trying to go backwards compatible doesn't work either. This is very frustrating as there is little to no support and the stupid thing doesn't work. As a software developer, this is very poor design. You guys should've known that upgrading would be a necessity in the future and made backwards compatibility a priority in order to prevent this feedback loop of being stuck. 

Now when I try to open the vault in the old version where it used to work it says the schema is too new, I have to import. WTF?!

**Comment by highOnBytes** [comment]
You can download an older version from mSigna here https://ciphrex.com/releases/win64/

I had the same issue when upgrading to a new PC, i installed newer mSigna and it was a no go. I just installed the older version so I could access my wallet. I will just send all my coins to my hardware wallet, upgrade mSigna and then load back what I want to trade with.
